lib/internal/webstreams/util.js

```diff
@@ -128,6 +129,24 @@ function transferArrayBuffer(buffer) {
   return res;
 }
 
+function isArrayBufferDetached(buffer) {
+  if (ArrayBufferGetByteLength(buffer) === 0) {
+    try {
+      new Uint8Array(buffer);
+    } catch {
+      return true;
+    }
```

Comment:
I'm not sure how slow this might really be but I guess it's fine for now because the Web Streams API is experimental.

@nodejs/streams wdyt?